src/api.py

```diff
@@ -242,11 +272,19 @@ def _generate_with_retry() -> Any:
             if functions:
                 kwargs = {**kwargs, "functions": functions}
 
-            logging.info("calling open ai chatcompletion create",
+            _, provider, _, _ = get_llm_provider(self.config.model)
+            if provider == "replicate":
+                # TODO This probably can be sidestepped by just using options for this.
```

Comment:
This was mentioned in the description for this PR, but we should move most of the LLM options to our options config, IMO.